tests/common-tests.h

```diff
@@ -4755,6 +4881,123 @@ TEST_CASE_METHOD(common_tests, "XML", "[core][xml]")
     }
 }
 
+// Tha same test as above, but using vectors of xml_type values.
+TEST_CASE_METHOD(common_tests, "XML vector", "[core][xml][vector]")
+{
+    soci::session sql(backEndFactory_, connectString_);
+
+    if (sql.get_backend_name() != "firebird" &&
+        sql.get_backend_name() != "odbc" &&
+        sql.get_backend_name() != "postgresql")
+    {
+        WARN("Vector of XML types not supported by the database, skipping the test.");
+        return;
+    }
+
+    auto_table_creator tableCreator(tc_.table_creator_xml(sql));
+    if (!tableCreator.get())
+    {
+        WARN("XML type not supported by the database, skipping the test.");
+        return;
+    }
+
+    std::vector<int> id(2);
+    id[0] = 1;
+    id[1] = 1; // Use the same ID to select both objects by ID.
```

Comment:
I don't think we can rely on the insertion and query order being the same. I.e. I think we need to use different IDs here and then add `order by id` clause to the select below. I'll do it, unless you tell me I'm missing something here.